app/views/recipes/show.html.erb

```diff
@@ -4,6 +4,17 @@
   <ul>
     <li><%= @recipe.ingredients %></li>
     <li><%= @recipe.instructions %></li>
+    pictures:
+    <% if @recipe.pictures.attached? %>
+      <li>
+        <p>
+        <strong>Pictures:</strong>
+        <br>
+        <%= @recipe.pictures.each do |picture| %>
+          <%= image_tag picture %>
+      <% end %>
+      </li>
+        </p>
   </ul>
-  <%= button_to t('.add_picture'), recipe_pictures_path(@recipe) %>
+<% end %>
```

Comment:
These tags seem mismatched. Is the generated HTML valid?

Reply:
I took a look at the whole file, and I am thinking that things would render more smoothly if the `</ul>` came _after_ the `<% end %>`. It appears as though if there are no pictures, then the closing tag wouldnt render either.